Documentation/design-docs/corefx-testing-framework.md

```diff
@@ -0,0 +1,53 @@
+The test scripts start and run multiple instances of the xunit runner in parallel. The test results are output under the `bin\Logs` folder and laterparsed by the CI machines.
+
+## Test binaries
+
+The binaries used to run the tests are stored in Azure Blob Storage.
+The current layout of blob storage is as follows – the root of the blob contains three folders for each OS. Each contains archived test binaries (i.e. the output of running `build-test` in CoreFX) and a .json file with a list of tests and their respective URLs. The json schema follows Helix's test list definitions, so that the test runs are compatible with Helix URLs.  
+
+To update them you need access to the secure connection string. Once you have the connection string available, you can use either Azure CLI, Azure Storage Explorer or the Blob Storage API.
+
+# Limitations and Future Development
+
+## Full architecture/OS matrix support
+
+This is the big lacking feature at the moment.
+
+## Exclusion list
+
+Defining all exclusions in a single file should be fairly straightforward. The schema is meant to be easily extensible - adding fields for exclusions on specific architectures and platforms would require a small change in how we generate `.rsp` files for the xunit test runner.
+
+## Blob Storage
+
+The current layout of blob storage is inelegant and not easy to support. We can define two potential paths for improvement:
+
++ Remove the .txt files with test URLs. Add a simple request handler between the blob storage and the testing framework to resolve URLs automatically for every supported platform/OS combo. This is where the binary refresh logic can also live – i.e. every time we update the CoreFX reference in CoreCLR, maestro-bot can send a request to refresh binaries. Then we can use helper scripts to generate test exclusions from failed tests so that CI is green while bugs are triaged.
+
++ The blob storage features above would duplicate some Helix functionality. The test list in each blob storage folder copies Helix’s JSON format. This was done intentionally to be compatible with the produced test payloads. We can use a Helix API to integrate test binary refresh with something similar to dotnet-maestro-bot and copy test binaries from a Helix run on a regular interval.
+
+## xUnit Console Runner
+
+Currently we depend on xunit.netcore.console.exe (our own implementation of an xUnit console runner) being present in the test payload. With the efforts going on in [the Arcade transition project](https://github.com/dotnet/corefx/projects/3) changes to the CoreFX test build format might happen, in which case we would need to start maintaining our one CoreFX console runner. If the binaries maintain backwards-compatibility (extremely likely) with the console runner, then we can simply call an instance of xunit.netcore.console.exe from the test host directory.
```

Comment:
I do not think we want to maintain `xunit.netcore.console.exe` when the rest of the world moved on. What is the problem with using the same runner as CoreFX with Arcade?

Reply:
Fine-grained test exclusions. If the version of xUnit used is high enough to pick up [the exclusion changes](https://github.com/xunit/xunit/pull/1734) , then we're in the clear. Otherwise, there has to be another mechanism of excluding single test cases.

Reply:
It would be nice to capture this information in the doc.